### PR TITLE
More enum work

### PR DESCRIPTION
**Pull request by youknowone**

### CODE REVIEW

File: crates/ruff_python_formatter/src/builders.rs [hunk 1]
```diff
@@ -36,7 +36,7 @@ pub struct LeadingComments<'a> {
 impl Format<ASTFormatContext<'_>> for LeadingComments<'_> {
     fn fmt(&self, f: &mut Formatter<ASTFormatContext<'_>>) -> FormatResult<()> {
         for comment in self.comments {
-            if matches!(comment.relationship, Relationship::Leading) {
+            if comment.relationship.is_leading() {
                 if let TriviaKind::OwnLineComment(range) = comment.kind {
```
Comment (youknowone) [inline crates/ruff_python_formatter/src/builders.rs:40]: This is up to preference, but if you want, these changes are also possible.

```suggestion
                if let Some(range) = comment.kind.own_line_comment() {
```

File: crates/ruff_cli/src/main.rs [hunk 2]
```diff
@@ -205,7 +205,7 @@ fn check(args: CheckArgs, log_level: LogLevel) -> Result<ExitStatus> {
     let printer = Printer::new(format, log_level, autofix, printer_flags);
 
     if cli.watch {
-        if !matches!(autofix, fix::FixMode::None) {
+        if !autofix.is_none() {
```
Comment (not-my-profile) [inline crates/ruff_cli/src/main.rs:208]: I actually consider this to be less readable. This now looks like `autofix` is an `Option`.

Comment (charliermarsh) [reply crates/ruff_cli/src/main.rs:208]: I guess `None` is arguably not a great enum variant name anyway?

File: crates/ruff_python_formatter/src/builders.rs (hunk 1 above)
Comment (charliermarsh) [reply crates/ruff_python_formatter/src/builders.rs:40]: I prefer what we have now.

File: crates/ruff_cli/src/main.rs (hunk 2 above)
Comment (youknowone) [reply crates/ruff_cli/src/main.rs]: I removed this commit

Comment (not-my-profile) [reply crates/ruff_cli/src/main.rs]: Thanks. I don't think there's any problem with having `None` as an enum variant name.

**Pull request merged**